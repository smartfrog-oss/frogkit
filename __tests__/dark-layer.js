import { mount } from 'vue-test-utils'
import DarkLayer from '@/components/dark-layer'

const slots = { default: '<span>Some text here</span>' }
const propsData = {show: true}

describe('Dark Layer', () => {
  it('should render component and match snapshot', () => {
    const wrapper = mount(DarkLayer, {slots, propsData})
    expect(wrapper.html()).toMatchSnapshot()
  })
  it('Dark Layer should emit click event when clicked', () => {
    const stub = jest.fn()
    const wrapper = mount(DarkLayer, {slots, propsData})
    wrapper.vm.$on('click', stub)
    wrapper.trigger('click')
    expect(stub).toBeCalled()
  })
})
