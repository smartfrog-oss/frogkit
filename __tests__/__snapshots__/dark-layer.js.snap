// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dark Layer should render component and match snapshot 1`] = `
<section class="fk-dark-layer">
  <flex column="" align="center" justify="center" grow=""><span>Some text here</span></flex>
</section>
`;
